python/pyspark/sql/decorators.py

```diff
@@ -0,0 +1,25 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.sql.types import StringType
+from pyspark.sql.functions import udf as _udf
+
+
+def udf(returnType=StringType()):
```

Comment:
OK, I moved decorator to `sql.functions` as `udf_decorator`. _Explicit is better than implicit_, right? Nevertheless I am reluctant to further attempts to merge it with the main `udf` function.

Technically It is possible, but it will require all kinds of special cases. It would be manageable for now, when we only support `returnType`, but if we ever decide to add further arguments (like `name`) it would be a mess. Especially when project is committed to supporting legacy Python versions. Otherwise we could just use `keyword` only arguments:

    def udf(func=None, *, returnType=StringType()):
        ... 

We could mimic that to some extent, without using backwards compatibility, by using keyword arguments:

    def udf(func=None, **kwargs):
        returnType = kwargs.get("returnType", StringType())
        ...

but I don't think it is worth loosing explicit argument list.